### DISCUSSION TITLE
Computed field based on elixir code instead of SQL

### DISCUSSION THREAD

**Discussion by phortx** [naymspace/backpex#559] [Q&A]
Hi,

I'm currently migrating from [Kaffy](https://github.com/aesmail/kaffy). I have several calculated fields, which base on business rules that are implemented in elixir. Kaffy supports this via passing a lambda to `value`.

As far as I understand, computed fields in Backpex are currently only possible via `select: dynamic()` which extends the query.
Is there also a way to display a computed field based on elixir code? For example:

```elixir
def fields do
  [
    free_percent: %{
      module: Backpex.Fields.Value,
      label: "Free usage this month",
      select: fn user -> to_string(Subscriptions.determine_quota(user).free_percent <> "%") end
    },
  ]
end
```

**Comment by Flo0807** [comment]
Hey,

the easiest way to achieve this is by defining a custom `render` function.

```elixir
def fields do
  [
    free_percent: %{
      module: Backpex.Fields.Text,
      label: "Free usage this month",
      render: fn assigns ->
        # user = ...
        free_percent = Subscriptions.determine_quota(user).free_percent <> "%"
        
        assigns = assign(assigns, :label, free_percent)
        
        ~H"""
        <p><%= @label %></p>
        """
      end
    },
  ]
end
```

in the custom `render` function, you have the flexibility to display whatever value you want to show.

Note that it sometimes may be better to use the `select: dynamic()` option because then your field can be orderable and searchable. In the above example, you'd have to turn `orderbale` to false because the ordering on database level would not make sense.

**Comment by phortx** [comment]
That's what I've been looking for, thanks!
